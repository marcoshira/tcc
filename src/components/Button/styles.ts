import styled, { css } from 'styled-components';

export const Wrapper = styled.button`
  ${({ theme }) => css`
    padding: ${theme.spacings.small} ${theme.spacings.xsmall};
    color: ${theme.colors.white};
    background: ${theme.colors.primaryColor};
    border-radius: 7px;
    box-shadow: 0px 1px 9px ${theme.colors.primaryColor};
    font-size: ${theme.font.sizes.ltmedium};
    font-weight: 500;
    margin-bottom: 4px;
    transition: ${theme.transitions.faster};

    &:disabled {
      background: ${theme.colors.grey};
      color: ${theme.colors.white};
      border: 2px solid ${theme.colors.grey};
      transition: none;

      &:hover {
        background: ${theme.colors.grey};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.grey};
        cursor: not-allowed;
      }
    }
    &:hover {
      cursor: pointer;
      background: inherit;
      color: ${theme.colors.primaryColor};
    }

    @media ${theme.media.gteMedium} {
      /* width: 33%; */
    }
  `}
`;
