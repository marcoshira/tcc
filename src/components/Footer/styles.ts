import styled, { css } from 'styled-components';

export const Footer = styled.div`
  ${({ theme }) => css`
    background: ${theme.colors.primaryColor};
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    color: white;
    padding: ${theme.spacings.small} 0 ${theme.spacings.medium};

    p {
      margin: 0;
      font-size: 20px;
    }
    a {
      font-size: 18px;
      text-decoration: none;
    }

    @media ${theme.media.gteMedium} {
      padding: ${theme.spacings.small} 8% ${theme.spacings.large};
      flex-direction: row;
      justify-content: space-around;
    }
  `}
`;

export const DevContainer = styled.div`
  ${({ theme }) => css`
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    margin-bottom: 20px;

    @media ${theme.media.gteMedium} {
      margin-bottom: 0px;
      p {
        margin-bottom: 20px;
        font-size: 25px;
      }
      a {
        font-size: 25px;
      }
    }
  `}
`;

export const IconsContainer = styled.div`
  ${({ theme }) => css`
    margin-top: 15px;
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    p {
      text-align: center;
    }
    svg {
      color: white;
      height: 40px;
    }
  `}
`;

export const IconsInnerContainer = styled.div`
  ${({ theme }) => css`
    padding: 0 20px;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    svg {
      color: white;
      height: 40px;
      margin-right: 10px;
    }
    a {
      text-decoration: none;
    }
  `}
`;
