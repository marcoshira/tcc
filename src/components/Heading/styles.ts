import styled, { css } from 'styled-components';
import { HeadingProps } from '.';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.huge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size ${theme.font.sizes.xlarge}
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1<HeadingProps>`
  ${({ theme, size, uppercase }) => css`
    color: white;
    margin: 0;
    margin-bottom: ${theme.spacings.small};
    padding: ${theme.spacings.large};
    background: ${theme.colors.primaryColor};
    text-align: center;
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
