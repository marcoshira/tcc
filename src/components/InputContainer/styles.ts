import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    box-shadow: 0px 1px 9px ${theme.colors.primaryColor};
    border-radius: 7px;
    margin-bottom: ${theme.spacings.small};
    width: 100%;
    transition: ${theme.transitions.slowest};

    @media ${theme.media.gteMedium} {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;
