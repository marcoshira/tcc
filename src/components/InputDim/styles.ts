import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    padding: 0 ${theme.spacings.large};
    margin: ${theme.spacings.medium} 0;
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    @media ${theme.media.gteMedium} {
      flex-direction: row;
      justify-content: space-around;
      /* width: 90%; */
    }

    label:first-of-type {
      width: 26%;
    }

    label:nth-of-type(2) {
      width: 31%;
    }
  `}
`;
