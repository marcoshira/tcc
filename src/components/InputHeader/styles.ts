import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    padding: 0px 0px;
    display: flex;
    justify-content: center;

    h2 {
      font-size: ${theme.font.sizes.medium};
      font-weight: 300;
    }
  `}
`;
