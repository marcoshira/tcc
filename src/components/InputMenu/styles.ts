import styled, { css } from 'styled-components';
import { InputMenuProps } from '.';

const frameSize = {
  small: (theme) => css`
    height: ${theme.frameSizes.ltxsmall};
    @media ${theme.media.gteMedium} {
      height: ${theme.frameSizes.xxsmall};
    }
  `,
  mediumSmall: (theme) => css`
    height: ${theme.frameSizes.small};
    @media ${theme.media.gteMedium} {
      height: ${theme.frameSizes.xxsmall};
    }
  `,
  medium: (theme) => css`
    height: ${theme.frameSizes.small};
    @media ${theme.media.gteMedium} {
      height: ${theme.frameSizes.ltxsmall};
    }
  `,
  big: (theme) => css`
    height: ${theme.frameSizes.gtsmall};
    @media ${theme.media.gteMedium} {
      height: ${theme.frameSizes.small};
    }
  `,
  huge: (theme) => css`
    height: ${theme.frameSizes.smallMedium};
    @media ${theme.media.gteMedium} {
      height: ${theme.frameSizes.gtsmall};
    }
  `,
};

export const Wrapper = styled.div<InputMenuProps>`
  ${({ theme, size }) => css`
    display: flex;
    justify-content: center;
    height: 0px;
    opacity: 0;

    transition: ${theme.transitions.faster};

    > div {
      display: none;
    }

    &.open {
      ${frameSize[size](theme)};
      opacity: 1;

      > div {
        display: flex;
      }
    }

    @media ${theme.media.gteMedium} {
      ${frameSize[size](theme)};
      opacity: 1;

      > div {
        display: flex;
      }
    }
  `}
`;
