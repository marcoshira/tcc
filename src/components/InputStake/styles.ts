import styled, { css } from 'styled-components';

export const WrapperPile = styled.div`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.large};
    width: 90%;
    flex-direction: column;
    justify-content: center;
  `}
`;

export const SubTypeContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.small};

    .control {
      display: flex;
      justify-content: end;
      position: relative;
      padding-left: 20px;
      margin-bottom: 5px;
      padding-top: 3px;
      cursor: pointer;
      font-size: ${theme.font.sizes.gtsmall};
    }
    .control input {
      position: absolute;
      z-index: -1;
      opacity: 0;
    }
    .control_indicator {
      position: absolute;
      top: 8px;
      left: 0px;
      height: 10px;
      width: 10px;
      background: #e6e6e7;
      border: 0px solid #000000;
    }
    .control:hover input ~ .control_indicator,
    .control input:focus ~ .control_indicator {
      background: #cccccc;
    }

    .control input:checked ~ .control_indicator {
      background: #0a1128;
    }
    .control:hover input:not([disabled]):checked ~ .control_indicator,
    .control input:checked:focus ~ .control_indicator {
      background: #0a1128;
    }
    .control input:disabled ~ .control_indicator {
      background: #e6e6e6;
      opacity: 0.6;
      pointer-events: none;
    }
    .control_indicator:after {
      box-sizing: unset;
      content: '';
      position: absolute;
      display: none;
    }
    .control input:checked ~ .control_indicator:after {
      display: block;
    }
    .control-radio .control_indicator {
      border-radius: 50%;
    }

    .control-radio .control_indicator:after {
      left: 7px;
      top: 7px;
      height: 0px;
      width: 0px;
      border-radius: 50%;
      background: #0a1128;
      transition: background 250ms;
    }
    .control-radio input:disabled ~ .control_indicator:after {
      background: #7b7b7b;
    }

    @media ${theme.media.gteMedium} {
      .control-group {
        display: flex;
        flex-direction: column;
        justify-content: space-between;
      }

      .flex {
        display: flex;
        flex-direction: row;
        justify-content: space-around;
      }

      .control {
        justify-content: left;
        margin: ${theme.spacings.xsmall} 0;
      }
    }
  `}
`;
