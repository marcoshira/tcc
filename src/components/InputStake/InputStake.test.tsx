import { fireEvent, screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { InputStake } from '.';

describe('<InputSoil />', () => {
  it('should render and match snapshot', () => {
    const { container } = renderTheme(<InputStake />);
    expect(
      screen.getByRole('radio', { name: 'Escavada em geral' }),
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
