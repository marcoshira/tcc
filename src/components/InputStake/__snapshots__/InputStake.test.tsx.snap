// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputSoil /> should render and match snapshot 1`] = `
.c0 {
  padding-bottom: 3.2rem;
  width: 80%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  margin: 2.4rem 0;
}

.c1 .control {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: end;
  -ms-flex-pack: end;
  justify-content: end;
  position: relative;
  padding-left: 20px;
  margin-bottom: 5px;
  padding-top: 3px;
  cursor: pointer;
  font-size: 1.6rem;
}

.c1 .control input {
  position: absolute;
  z-index: -1;
  opacity: 0;
}

.c1 .control_indicator {
  position: absolute;
  top: 8px;
  left: 0px;
  height: 10px;
  width: 10px;
  background: #e6e6e7;
  border: 0px solid #000000;
}

.c1 .control:hover input ~ .control_indicator,
.c1 .control input:focus ~ .control_indicator {
  background: #cccccc;
}

.c1 .control input:checked ~ .control_indicator {
  background: #0a1128;
}

.c1 .control:hover input:not([disabled]):checked ~ .control_indicator,
.c1 .control input:checked:focus ~ .control_indicator {
  background: #0a1128;
}

.c1 .control input:disabled ~ .control_indicator {
  background: #e6e6e6;
  opacity: 0.6;
  pointer-events: none;
}

.c1 .control_indicator:after {
  box-sizing: unset;
  content: '';
  position: absolute;
  display: none;
}

.c1 .control input:checked ~ .control_indicator:after {
  display: block;
}

.c1 .control-radio .control_indicator {
  border-radius: 50%;
}

.c1 .control-radio .control_indicator:after {
  left: 7px;
  top: 7px;
  height: 0px;
  width: 0px;
  border-radius: 50%;
  background: #0a1128;
  -webkit-transition: background 250ms;
  transition: background 250ms;
}

.c1 .control-radio input:disabled ~ .control_indicator:after {
  background: #7b7b7b;
}

@media (min-width:768px) {
  .c1 .control-group {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }

  .c1 .flex {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: space-around;
    -webkit-justify-content: space-around;
    -ms-flex-pack: space-around;
    justify-content: space-around;
  }

  .c1 .control {
    -webkit-box-pack: left;
    -webkit-justify-content: left;
    -ms-flex-pack: left;
    justify-content: left;
    margin: 0.8rem 0;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="control-group"
      >
        <div
          class="flex"
        >
          <label
            class="control control-radio"
          >
            Escavada em geral
            <input
              checked=""
              name="stake"
              type="radio"
            />
            <div
              class="control_indicator"
            />
          </label>
          <label
            class="control control-radio"
          >
            Escavada bentonita
            <input
              name="stake"
              type="radio"
            />
            <div
              class="control_indicator"
            />
          </label>
        </div>
        <div
          class="flex"
        >
          <label
            class="control control-radio"
          >
            Hélice contínua
            <input
              name="stake"
              type="radio"
            />
            <div
              class="control_indicator"
            />
          </label>
          <label
            class="control control-radio"
          >
            Raiz
            <input
              name="stake"
              type="radio"
            />
            <div
              class="control_indicator"
            />
          </label>
        </div>
        <div
          class="flex"
        >
          <label
            class="control control-radio"
          >
            Injetada sob altas pressões
            <input
              name="stake"
              type="radio"
            />
            <div
              class="control_indicator"
            />
          </label>
          <label
            class="control control-radio"
          >
            Franki
            <input
              name="stake"
              type="radio"
            />
            <div
              class="control_indicator"
            />
          </label>
        </div>
        <div
          class="flex"
        >
          <label
            class="control control-radio"
          >
            Metálica
            <input
              name="stake"
              type="radio"
            />
            <div
              class="control_indicator"
            />
          </label>
          <label
            class="control control-radio"
          >
            Pré-moldada
            <input
              name="stake"
              type="radio"
            />
            <div
              class="control_indicator"
            />
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
